pkg/personname/info_test.go

```diff
@@ -0,0 +1,403 @@
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Phonetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+			},
+			NoNullSeparators: false,
+			IsEmpty:          true,
+		},
+		// Empty no seps
+		{
+			Raw: "",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Ideographic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Phonetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+			},
+			NoNullSeparators: true,
+			IsEmpty:          true,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		// Test creating a new Info object.
+		t.Run(tc.Raw+"_new", func(t *testing.T) {
+			newInfo := Info{
+				Alphabetic:       tc.Expected.Alphabetic,
+				Ideographic:      tc.Expected.Ideographic,
+				Phonetic:         tc.Expected.Phonetic,
+				NoNullSeparators: tc.NoNullSeparators,
+			}
+
+			if tc.Raw != newInfo.String() {
+				t.Errorf(
+					"formatted string: expected '%v', got '%v'",
+					tc.Raw,
+					newInfo.String(),
+				)
+			}
+		})
+
+		// Test parsing a full PN value.
+		t.Run(tc.Raw+"_parse", func(t *testing.T) {
+
+			parsed, err := Parse(tc.Raw)
+			if err != nil {
+				t.Fatal("error parsing value:", err)
+			}
+
+			checkGroupInfo(
+				t,
+				tc.Expected.Alphabetic,
+				tc.Expected.Alphabetic.String(),
+				parsed.Alphabetic,
+				"Alphabetic",
+			)
+			checkGroupInfo(
+				t,
+				tc.Expected.Ideographic,
+				tc.Expected.Ideographic.String(),
+				parsed.Ideographic,
+				"Ideographic",
+			)
+			checkGroupInfo(
+				t,
+				tc.Expected.Phonetic,
+				tc.Expected.Phonetic.String(),
+				parsed.Phonetic,
+				"Phonetic",
+			)
+		})
+
+		// Test the .IsEmpty() method.
+		t.Run(tc.Raw+"_isEmpty", func(t *testing.T) {
+			newInfo, err := Parse(tc.Raw)
+			if err != nil {
+				t.Error("error parsing value:", err)
+				t.FailNow()
+			}
+
+			if tc.IsEmpty != newInfo.IsEmpty() {
+				t.Errorf(
+					".IsEmpty(): got '%v', expected '%v'",
+					tc.IsEmpty,
+					newInfo.IsEmpty(),
+				)
+			}
+		})
+	}
+}
+
+func TestParse_Err(t *testing.T) {
+	testCases := []struct {
+		Raw       string
+		ErrString string
+	}{
+		{
+			Raw: "===",
+			ErrString: "string contains to many segments for PN value:" +
+				" PN contains 4 groups. No more than 3 groups with" +
+				" '[Alphabetic]=[Ideographic]=[Phonetic]' format are allowed",
+		},
```

Comment:
typically I don't test the full ErrString, usually only that errors.Is() the underlying error (which is what the user would likely rely on in code for err handling). Up to you on if you want to test the formatting details.

Reply:
I like adding tests for error formatting. I consider helpful error messages to be a part of a library's feature set, even if it is a human-centric part.